src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverter.cs

```diff
@@ -67,7 +67,7 @@ internal virtual object CreateObject(JsonSerializerOptions options)
             throw new InvalidOperationException(SR.NodeJsonObjectCustomConverterNotAllowedOnExtensionProperty);
         }
 
-        internal abstract JsonPropertyInfo CreateJsonPropertyInfo(JsonTypeInfo parentTypeInfo);
+        internal abstract JsonPropertyInfo CreateJsonPropertyInfo(JsonTypeInfo declaringTypeInfo, Type propertyType, JsonSerializerOptions options);
```

Comment:
do we need to do all these refactoring? Implementing AttributeProvider is only around 40-50 lines changes

Reply:
The changes are not needed for `AttributeProvider`, I'm merely incuding it here since it would have been impossible to rebase on top of the refactoring.

The refactoring itself is motivated by changes required by the `IsExtensionData` property, which necessitates changes in how the PropertyCache is populated. Effectively, I want to make it so that the `PropertyCache` is only computed once at the `Configure()` stage (although this PR doesn't make that change yet).